pkg/queues/amqp_provider.go

```diff
@@ -0,0 +1,189 @@
+package queues
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/flightctl/flightctl/pkg/reqid"
+	"github.com/go-chi/chi/v5/middleware"
+	amqp "github.com/rabbitmq/amqp091-go"
+	"github.com/sirupsen/logrus"
+)
+
+type amqpProvider struct {
+	ctx     context.Context
+	cancel  context.CancelFunc
+	url     string
+	log     logrus.FieldLogger
+	wg      *sync.WaitGroup
+	queues  []*amqpQueue
+	stopped bool
+}
+
+func NewAmqpProvider(url string, log logrus.FieldLogger) Provider {
+	var wg sync.WaitGroup
+	ctx, cancel := context.WithCancel(context.Background())
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		<-ctx.Done()
+	}()
```

Comment:
feels a little strange for a constructor to start a go routine that just waits for ctx to be cancelled.

Reply:
The intention was that `Wait` can called on a provider even if there are no consumer loops.  I will try to avoid it.